tests/integration/test_examples.py

```diff
@@ -0,0 +1,95 @@
+from collections.abc import Callable
+
+from guppylang.decorator import guppy
+from guppylang.module import GuppyModule
+from guppylang.prelude.builtins import result
+from guppylang.prelude.quantum import qubit
+
+import guppylang.prelude.quantum as quantum
+from guppylang.prelude.quantum import h, cx, x, measure, rz, reset
+
+
+def test_random_walk_phase_estimation(validate):
+    # Implementation of the adaptive random walk phase estimation algorithm from
```

Comment:
We can use docstrings for this since #334

Reply:
The Guppy functions below already have docstrings (thanks btw!). Not sure if people usually add docstrings to pytest functions?

Reply:
Oops, i wasn't paying close enough attention to what kind of function i was looking at!

It just seemed odd to open a python function with a descriptive comment vs a docstring even if it's not library code 🤷 of course, i'm no expert on python conventions